botocore/session.py

```diff
@@ -164,6 +165,7 @@ def _register_components(self):
         self._register_endpoint_resolver()
         self._register_event_emitter()
         self._register_response_parser_factory()
+        self._register_bucket_region_cache()
```

Comment:
Thinking out loud:

a) If we think a general purpose cache is useful that can be shared amongst various clients then we can generalize the cache and not make it S3 specific.  Might be dangerous to let any arbitrary objects write to it.
b) We need to design a way to allow for service specific objects to be integrated into client creation in a more general purpose way.  It doesn't seem like S3 will be the last service that might need service specific objects.